cmd/config/config.go

```diff
@@ -1,82 +1,145 @@
 package main
 
 import (
-	"encoding/json"
 	"flag"
-	"log"
+	"fmt"
 	"os"
+	"slices"
+	"strings"
 
 	"github.com/thrasher-corp/gocryptotrader/common/file"
 	"github.com/thrasher-corp/gocryptotrader/config"
 )
 
-// EncryptOrDecrypt returns a string from a boolean
-func EncryptOrDecrypt(encrypt bool) string {
-	if encrypt {
-		return "encrypted"
-	}
-	return "decrypted"
-}
+var commands = []string{"upgrade", "encrypt", "decrypt"}
 
 func main() {
-	var inFile, outFile, key string
-	var encrypt bool
+	fmt.Println("GoCryptoTrader: config-helper tool")
+
 	defaultCfgFile := config.DefaultFilePath()
-	flag.StringVar(&inFile, "infile", defaultCfgFile, "The config input file to process.")
-	flag.StringVar(&outFile, "outfile", defaultCfgFile+".out", "The config output file.")
-	flag.BoolVar(&encrypt, "encrypt", true, "Whether to encrypt or decrypt.")
-	flag.StringVar(&key, "key", "", "The key to use for AES encryption.")
-	flag.Parse()
 
-	log.Println("GoCryptoTrader: config-helper tool.")
+	var in, out, key string
+	fs := flag.NewFlagSet("config", flag.ExitOnError)
+	fs.Usage = func() { usage(fs) }
+	fs.StringVar(&in, "infile", defaultCfgFile, "The config input file to process")
+	fs.StringVar(&out, "outfile", "[infile].out", "The config output file")
+	fs.StringVar(&key, "key", "", "The key to use for AES encryption")
+	_ = fs.Parse(os.Args[1:])
+
+	if out == "[infile].out" {
+		out = in + ".out"
+	}
+
+	switch parseCommand(fs) {
+	case "upgrade":
+		upgradeFile(in, out)
+	case "decrypt":
+		encryptWrapper(in, out, key, decryptFile)
+	case "encrypt":
+		encryptWrapper(in, out, key, encryptFile)
+	}
 
+	fmt.Println("Success! File written to " + out)
+}
+
+func upgradeFile(in, out string) {
+	if config.IsFileEncrypted(in) {
+		fatal("Cannot upgrade an encrypted file. Please decrypt first")
```

Comment:
Why can't you prompt for key? Why can't there be decrypted config in memory instead of having to save decryption to disk to continue?

```
go run . --infile="/Users/scottg/.gocryptotrader/config.json.out" --key="1" upgrade
GoCryptoTrader: config-helper tool
Cannot upgrade an encrypted file. Please decrypt first
exit status 2
```

Reply:
We could.
I'm trying to decrease footprint.
There's a lot of places where we've added features that aren't necessary, and it's resulted in a very wide codebase.
This is one of the places where the number of times a user will run upgrade _manually_ is limited, and supporting decryption isn't free.
Note: I *really* want to break encryption out of this entirely as soon as hell.
Secrets belong in a secret store merged into config, not mixed in with config.
Config merging and secret mounting was one of the most painful areas of service-orienting gct for cloud deployment for me originally.

So: Do you need me to support decryption for upgrades? :)

Reply:
>This is one of the places where the number of times a user will run upgrade manually is limited

? I can only review the code and the pull request text in front of me. Is the goal here to upgrade user's config in place on run in future or something?

I have to decrypt my config, upgrade my config with a new infile and outfile, encrypt it again with a new infile and outfile and replace my original config to upgrade my config then delete the old ones because I don't want my keys to be unencrypted at rest. Below is with an encrypted config
```
go run . --key="1" --infile="/Users/DOGE/.gocryptotrader/config.json.out2" upgrade
GoCryptoTrader: config-helper tool
2024/11/14 11:17:39 Running upgrade to config version 0
2024/11/14 11:17:39 Running upgrade to config version 1
2024/11/14 11:17:39 Running upgrade to config version 2
2024/11/14 11:17:39 Version management finished
Success! File written to /Users/scottg/.gocryptotrader/config.json.out2.out
```

I'll give you the benefit of the doubt and not make you do it then. Right now its an elaborate process prone to error due to the level of manual interventions and I was just suggesting ways to improve it.

Reply:
Yeah, okay, I see your point.
I'll make it so.

Reply:
Fixed https://github.com/gbjk/gocryptotrader/commit/1b498f3bb

Upgrade now supports encryption.

Reply:
Excellent updates, thank you 🚀